src/common.c

```diff
@@ -2951,30 +2951,57 @@ int stlink_erase_flash_page(stlink_t *sl, stm32_addr_t flashaddr) {
   return check_flash_error(sl);
 }
 
+int stlink_erase_flash_section(stlink_t *sl, stm32_addr_t base_addr, size_t size) {
+  if (base_addr < sl->flash_base || (base_addr + size) > (sl->flash_base + sl->flash_size)) {
+    ELOG("Invalid address or flash size\n");
+    return (-1);
+  }
+
+  stm32_addr_t addr = sl->flash_base;
+
+  // Make sure the requested address is aligned with the beginning of a page
+  while (addr < base_addr) {
+    addr += stlink_calculate_pagesize(sl, addr);
+  }
+  if (addr != base_addr) {
```

Comment:
This line denies me to erase any sector except the first.

Reply:
Yeah, can we find a proper solution for that?

Reply:
Are you sure the address you used was aligned with the beginning of a page ?
On my H7 (with 128kB sectors) I can erase `0x08000000`, `0x08020000`, `0x08040000` and so on. Is `stlink_calculate_pagesize` supporting your flash size ?

Reply:
IMHO alignment is even a basic requirement for some memory regions according to common ARM reference guides (e.g. The Definitive Guide to ARM® Cortex®-M3 and Cortex®-M).
I found this as a direct reference as well: https://wiki.segger.com/Open_Flashloader#Example

Reply:
Yeah but that's what I'm doing here: I'm iterating through the pages to find the one we want to erase, and I check if the address is equal to the first address of the page.
Hence the question to @slyshykO regarding the parameters / chip he used, because I've tested this on my H7 (the only board I have at the moment) and I was able to erase starting from the 2nd, 3rd etc page.

Reply:
e.g. : 
```
./build/Release/bin/st-flash --connect-under-reset --reset --area=main erase 0x08060000 128k
st-flash 1.7.0-133-gf277fdb
2022-01-12T09:31:57 INFO common.c: H74x/H75x: 128 KiB SRAM, 2048 KiB flash in at least 128 KiB pages.
-> Flash page at 0x8060000 erased (size: 0x20000)
```

```
./build/Release/bin/st-flash --connect-under-reset --reset --area=main erase 0x08080000 256k
st-flash 1.7.0-133-gf277fdb
2022-01-12T09:32:17 INFO common.c: H74x/H75x: 128 KiB SRAM, 2048 KiB flash in at least 128 KiB pages.
-> Flash page at 0x8080000 erased (size: 0x20000)
-> Flash page at 0x80a0000 erased (size: 0x20000)
```

Reply:
Ping @slyshykO

Reply:
@slyshykO Does this work now for you with aligning?